weave-js/src/components/PagePanelComponents/Home/Browse3/typeViews/File/FileView.tsx

```diff
@@ -0,0 +1,251 @@
+import {Button} from '@wandb/weave/components/Button';
+import {Tooltip} from '@wandb/weave/components/Tooltip';
+import React, {useCallback, useEffect, useState} from 'react';
+
+import {convertBytes} from '../../../../../../util';
+import {Icon, IconName} from '../../../../../Icon';
+import {LoadingDots} from '../../../../../LoadingDots';
+import {TailwindContents} from '../../../../../Tailwind';
+import {CustomLink} from '../../pages/common/Links';
+import {useWFHooks} from '../../pages/wfReactInterface/context';
+import {CustomWeaveTypePayload} from '../customWeaveType.types';
+import {PDFView} from './PDFView';
+
+const ICON_MAP: Record<string, IconName> = {
+  'application/json': 'job-program-code',
+  'text/csv': 'table',
+  'text/html': 'job-program-code',
+  'text/xml': 'job-program-code',
```

Comment:
Maybe use the `IconNames` const values instead?

Reply:
Is there a reason you prefer the more verbose style?

Reply:
Not any safe type-wise because its a `Record<string, IconName>` but I will change if you think it improves readability.

Reply:
Yes, I realize `IconName` is just as safe. As you wish, no obligation. I personally don't like loose strings 😁